nucliadb_telemetry/nucliadb_telemetry/errors.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from typing import Any, ContextManager, Optional
+
+import pydantic
+
+try:
+    import sentry_sdk
+    from sentry_sdk import Scope
+    from sentry_sdk.integrations.logging import LoggingIntegration
+
+    SENTRY = True
+except ImportError:  # pragma: no cover
+    Scope = LoggingIntegration = sentry_sdk = None  # type: ignore
+    SENTRY = False
+
+
+def capture_exception(error: BaseException) -> Optional[str]:
+    if SENTRY:
+        return sentry_sdk.capture_exception(error)
+    return None
+
+
+class NoopScope:
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        ...
+
+    def set_extra(self, key: str, value: Any) -> None:
+        ...
+
+
+def push_scope(**kwargs: Any) -> ContextManager[Scope]:
+    if SENTRY:
+        return sentry_sdk.push_scope(**kwargs)
+    else:
+        return NoopScope()  # type: ignore
+
+
+class ErrorHandlingSettings(pydantic.BaseSettings):
+    sentry_url: Optional[str] = None
+    environment: str = pydantic.Field(
+        "local", env=["environment", "running_environment"]
+    )
+    logging_integration: bool = False
+
+
+def setup_error_handling(version: str) -> None:
+    settings = ErrorHandlingSettings()
+
+    if settings.sentry_url:
+        enabled_integrations: list[Any] = []
+
+        if settings.logging_integration:
+            sentry_logging = LoggingIntegration(
+                level=logging.CRITICAL, event_level=logging.CRITICAL
```

Comment:
Should level be a parameter? If I am not wrong, for some components we are receiving logger.ERRORs on sentry right now.

Reply:
This was a straight copy of the current implementation. I think I'll leave for now and wait until we actually need something else.